examples/all-clusters-app/all-clusters-common/gen/endpoint_config.h

```diff
@@ -15,365 +15,392 @@
+              ZAP_CLUSTER_MASK(SERVER) | ZAP_CLUSTER_MASK(INIT_FUNCTION),                                                          \
+              chipFuncArrayOnOffServer }, /* Endpoint: 2, Cluster: On/off (server) */                                              \
     }
 
-// Endpoint types
-#define GENERATED_ENDPOINT_TYPES                                                                                                   \
-    {                                                                                                                              \
-        { (EmberAfCluster *) &(generatedClusters[0]), 23, 147 }, { (EmberAfCluster *) &(generatedClusters[23]), 1, 3 },            \
-    }
+#define ZAP_CLUSTER_INDEX(index) ((EmberAfCluster *) (&generatedClusters[index]))
 
-// Cluster manufacturer codes
-#define GENERATED_CLUSTER_MANUFACTURER_CODES                                                                                       \
-    {                                                                                                                              \
-        {                                                                                                                          \
-            0x00, 0x00                                                                                                             \
-        }                                                                                                                          \
-    }
-#define GENERATED_CLUSTER_MANUFACTURER_CODE_COUNT (0)
-
-// Attribute manufacturer codes
-#define GENERATED_ATTRIBUTE_MANUFACTURER_CODES                                                                                     \
+// This is an array of EmberAfEndpointType structures.
+#define GENERATED_ENDPOINT_TYPES                                                                                                   \
     {                                                                                                                              \
-        {                                                                                                                          \
-            0x00, 0x00                                                                                                             \
-        }                                                                                                                          \
+        { ZAP_CLUSTER_INDEX(0), 20, 415 }, { ZAP_CLUSTER_INDEX(20), 2, 7 },                                                        \
     }
-#define GENERATED_ATTRIBUTE_MANUFACTURER_CODE_COUNT (0)
 
 // Largest attribute size is needed for various buffers
-#define ATTRIBUTE_LARGEST (8)
+#define ATTRIBUTE_LARGEST (254)
+
 // Total size of singleton attributes
-#define ATTRIBUTE_SINGLETONS_SIZE (6)
+#define ATTRIBUTE_SINGLETONS_SIZE (0)
 
 // Total size of attribute storage
-#define ATTRIBUTE_MAX_SIZE 150
+#define ATTRIBUTE_MAX_SIZE (422)
```

Comment:
It still looks weird that `ATTRIBUTE_MAX_SIZE' goes to `422`. I think you already answered me but I can't find your answer anymore. Can you explain it to me again ? (sorry).

Reply:
Ah right. I remember it cames mostly "from compensation text". That said I don't remember why it is added ?

Reply:
It is because the attribute is enabled in the configuration and the maximum string length of said attribute is 256 ish.